### PR TITLE
Added react query

### PR DESCRIPTION
**Pull request by fero1xd**
This pr adds react query for data fetching

### CODE REVIEW

**Comment by fero1xd** [comment]
yea, but i thought npm is being used as a package manager in this repo

File: pnpm-lock.yaml
Comment (manjotsk) [inline pnpm-lock.yaml:1]: Thank you for your contribution. Can you reinstate this file?

**Comment by fero1xd** [comment]
u want me to add it back ?

File: src/pages/ChessNews.tsx [hunk 1]
```diff
@@ -1,103 +1,98 @@
...
-
     // Check for token in URL or localStorage
     const searchParams = new URLSearchParams(location.search);
     const token = searchParams.get('token') || localStorage.getItem('token');
     setShowSidebar(!!token);
   }, [location.search]);
 
   return (
-    <div className="flex h-screen bg-gray-900 text-white">
+    <div className='flex h-screen bg-gray-900 text-white'>
```
Comment (manjotsk) [inline src/pages/ChessNews.tsx:23]: @Bot-Rakshit  I guess we need to set up eslint settings to honor uniformity.

File: pnpm-lock.yaml
Comment (fero1xd) [reply pnpm-lock.yaml:1]: Yea i have added this back

File: src/pages/ChessNews.tsx (hunk 1 above)
Comment (fero1xd) [reply src/pages/ChessNews.tsx:23]: This is prettier doing this, ig there should be a root prettier config file

**Comment by manjotsk** [comment]
@fero1xd Seems like another version of nodejs is being used in your dev env. there are a lot of lockfile changes. Also there are some conflicts! I guess @Bot-Rakshit developed this using v22.x.x

**Comment by fero1xd** [comment]
@Bot-Rakshit let me know the exact version of node that you are using and i will fix the other conflicts

**Comment by manjotsk** [comment]
@fero1xd He is using v22.0.0 
image

**Comment by fero1xd** [comment]
lock file conflicts are horrible.. should be good now

**Pull request closed without merging**